dev/docker/iceberg-rest-server/Dockerfile

```diff
@@ -0,0 +1,28 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+FROM openjdk:8-jdk-buster
```

Comment:
Should use a higher JDK such as 11 or 17?

Reply:
Gravitino docker image use JDK8 too, change it to JDK17 too?

Reply:
It's not a big issue to use JDK8 for now, I guess.

Reply:
It's better to use a higher JDK since the JDK8 is out of style.

Reply:
Let's change to 17.

Reply:
ok, updated to jdk17

Reply:
updated and verified JDK17 for Gravitino server and Iceberg REST server